src/components/UserFeed.jsx

```diff
@@ -1,69 +1,126 @@
+            !myUserFollowing.includes(recipe.username)
+          );
+          setRecipes(filteredRecipes);
+          setLoading(false);
+        })
+        .catch((error) => { 
+          console.error("Error al obtener recetas:", error); 
+          setLoading(false); 
+        });
+    }
+};
+
 
   return (
-    <Container className="pb-5 pt-3">
-      <Row>
-        {recipes && recipes.length > 0 ? (
-          recipes.map((recipe) => (
-            <Col sm={12} md={6} xl={4}>
-              <CSSTransition in={true} timeout={500} classNames="slideUp" appear>
-                <Link key={recipe._id} to={`/RecipeDetail/${recipe._id}`} className="text-decoration-none">
-                  <Card className="mt-5 shadow transition-03s" id="recipes-list">
-                    <Card.Img className="object-fit-cover" variant="top" src={recipe.main_image ?? gyoza} height={300}/>
-                    <Card.Body>
-                      <Card.Title className="overflow-hidden text-nowrap pb-1">{recipe.name}</Card.Title>
-                        <h5><Image src={chefIcon} style={{height:'24px', width: '24px'}} fluid/> {Array(recipe.difficulty || 0).fill().map((_, index) => (
-                            <span key={index} className="fs-5 ms-1 text-center"><StarFill style={{color: 'gold'}}></StarFill></span>
-                          ))}</h5>
-                        <div className="d-flex">
-                          <StarFill className="mx-1 mt-1" style={{color: 'red'}}></StarFill>
-                          <span>{recipe?.average_rating?.toFixed(1) || 0}</span>
-                        </div>
-                    </Card.Body>
-                    <Card.Footer>By {recipe?.username}</Card.Footer>
-                  </Card>
-                </Link>
-              </CSSTransition>
-            </Col>
-          ))
-        ) : ( <div className="alert alert-warning" role="alert">There are currently no recipes</div> )
-        }
-      </Row>
+    <Container>
+      <Row></Row>
+      <div className="d-flex justify-content-center mt-3">
+      <ButtonGroup className="mb-3">
+        <Button 
+          style={feedType === "forYou" ? { backgroundColor: '#FFC1AC', borderColor: '#FFC1AC', color: '#000' } : null}
+          variant={feedType === "forYou" ? "light" : "light"} 
+          onClick={() => setFeedType("forYou")}
+        >
+          For You
+        </Button>
+        <Button 
+          style={feedType === "following" ? { backgroundColor: '#FFC1AC', borderColor: '#FFC1AC', color: '#000' } : null}
+          variant={feedType === "following" ? "light" : "light"} 
+          onClick={() => setFeedType("following")}
+        >
+          Following
+        </Button>
+      </ButtonGroup>
+
+      </div>
+      {loading ? (
+        <Container className="d-flex justify-content-center align-items-center min-vh-100">
+          <Spinner animation="border" variant="secondary"/>
+        </Container>
+      ) : ( 
+        <RecipeList recipes={recipes} /> 
```

Comment:
After applying the previous changes you can just change the prop
```suggestion
        <RecipeList recipes={() => {
          if (feedType === "following") {
            return recipes;
          } else if (feedType === "forYou") {
            return followingRecipes;
          }
        }} /> 
```

Reply:
I have made it in a single method to encapsulate the code.